_posts/2019-06-25-dynamic_linear_models_tfprobability/dynamic_linear_models_tfprobability.Rmd

```diff
@@ -0,0 +1,805 @@
+Besides introducing the [dlm](https://cran.r-project.org/web/packages/dlm/index.html) package, this book provides a nice introduction to the ideas behind DLMs in general.
+
+To illustrate dynamic linear regression, the authors feature a dataset, originally from Berndt(1991)[@Berndt-1991] that has monthly returns, collected from January 1978 to December 1987, for four different stocks, the 30-day Treasury Bill -- standing in for a _risk-free_ asset --, and the value-weighted average returns for all stocks listed at the New York and American Stock Exchanges, representing the overall _market returns_. 
+
+Let's take a look.
+
+```{r}
+# As the data does not seem to be available at the address given in Petris et al. any more,
+# we put it on the blog for download
+file <- download.file(
+  "https://github.com/rstudio/tensorflow-blog/tree/master/docs/posts/2019-06-25-dynamic-linear-models-tfprobability/data/capm.txt",
+  "capm.txt")
+df <- read_table(
+  filename,
+  col_types = list(
+    col_date(format = "%Y.%m"),
+    col_double(),
+    col_double(),
+    col_double(),
+    col_double(),
+    col_double(),
+    col_double()
+  )
+) %>%
+  rename(month = X1)
+
+df %>% glimpse()
+```
+
+```
+Observations: 120
+Variables: 7
+$ month  <date> 1978-01-01, 1978-02-01, 1978-03-01, 1978-04-01, 1978-05-01, 19…
+$ MOBIL  <dbl> -0.046, -0.017, 0.049, 0.077, -0.011, -0.043, 0.028, 0.056, 0.0…
+$ IBM    <dbl> -0.029, -0.043, -0.063, 0.130, -0.018, -0.004, 0.092, 0.049, -0…
+$ WEYER  <dbl> -0.116, -0.135, 0.084, 0.144, -0.031, 0.005, 0.164, 0.039, -0.0…
+$ CITCRP <dbl> -0.115, -0.019, 0.059, 0.127, 0.005, 0.007, 0.032, 0.088, 0.011…
+$ MARKET <dbl> -0.045, 0.010, 0.050, 0.063, 0.067, 0.007, 0.071, 0.079, 0.002,…
+$ RKFREE <dbl> 0.00487, 0.00494, 0.00526, 0.00491, 0.00513, 0.00527, 0.00528, …
+```
+
+```{r}
+df %>% gather(key = "symbol", value = "return", -month) %>%
+  ggplot(aes(x = month, y = return, color = symbol)) +
+  geom_line() +
+  facet_grid(rows = vars(symbol), scales = "free")
+```
+
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body-outset", fig.cap = "Monthly returns for selected stocks; data from Berndt (1991)."}
+knitr::include_graphics("images/capm_overview.png")
+```
+
+
+The [Capital Asset Pricing Model](https://en.wikipedia.org/wiki/Capital_asset_pricing_model) then assumes a linear relationship between the excess returns of an asset under study and the excess returns of the market. For both, _excess returns_ are obtained by subtracting the returns of the chosen _risk-free_ asset; then, the scaling coefficient between them reveals the asset to either be an "aggressive" investment (slope > 1: changes in the market are amplified), or a conservative one (slope < 1: changes are damped).
+
+Assuming this relationship does not change over time, we can easily use `lm` to illustrate this. Following Petris et al. in zooming in on IBM as the asset under study, we have
```

Comment:
```suggestion
Assuming this relationship does not change over time, we can use `lm` to illustrate this. Following Petris et al. in zooming in on IBM as the asset under study, we have
```

Reply:
You'll demonstrate the ease with which you can do this. (I try to remove the likes of "easy" and "simple" from technical writing, since, for someone, it's inevitably not).